src/foremast/pipeline/create_pipeline_datapipeline.py

```diff
@@ -0,0 +1,136 @@
+#
+#   Copyright 2016 Gogo, LLC
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""Create Pipelines for Spinnaker."""
+import collections
+import json
+import logging
+import os
+from pprint import pformat
+
+import requests
+
+from ..consts import API_URL
+from ..exceptions import SpinnakerPipelineCreationFailed
+from ..utils import get_details, get_properties, get_template
+from .clean_pipelines import clean_pipelines
+from .construct_pipeline_block_datapipeline import construct_pipeline_block_datapipeline
+from .create_pipeline import SpinnakerPipeline
+from .renumerate_stages import renumerate_stages
+
+
+class SpinnakerPipelineDataPipeline(SpinnakerPipeline):
+    """Manipulate Spinnaker Pipelines.
+
+    Args:
+        app (str): Application name.
+        trigger_job (str): Jenkins trigger job.
+        base (str): Base image name (i.e: fedora).
+        prop_path (str): Path to the raw.properties.json.
+    """
+
+    def render_wrapper(self, region='us-east-1'):
+        """Generate the base Pipeline wrapper.
+
+        This renders the non-repeatable stages in a pipeline, like jenkins, baking, tagging and notifications.
+
+        Args:
+            region (str): AWS Region.
+
+        Returns:
+            dict: Rendered Pipeline wrapper.
+        """
+        base = self.settings['pipeline']['base']
+
+        if self.base:
+            base = self.base
+
+        email = self.settings['pipeline']['notifications']['email']
+        slack = self.settings['pipeline']['notifications']['slack']
+        deploy_type = self.settings['pipeline']['type']
+        provider = 'aws'
+        pipeline_id = self.compare_with_existing(region=region)
+
+        data = {
+            'app': {
+                'appname': self.app_name,
+                'base': base,
+                'deploy_type': deploy_type,
+                'environment': 'packaging',
+                'region': region,
+                'triggerjob': self.trigger_job,
+                'email': email,
+                'slack': slack,
+            },
+            'id': pipeline_id
+        }
+
+        self.log.debug('Wrapper app data:\n%s', pformat(data))
+
+        wrapper = get_template(
+            template_file='pipeline/pipeline_wrapper.json.j2',
+            data=data)
+
+        return json.loads(wrapper)
+
+    def create_pipeline(self):
+        """Main wrapper for pipeline creation.
+        1. Runs clean_pipelines to clean up existing ones
```

Comment:
Because I am afraid of increasing scope and time of this feature by refactoring. What is there currently works and is consistent with the other code and this PR is already long enough